boot/boot_serial/src/boot_serial.c

```diff
@@ -312,10 +313,13 @@ bs_upload(char *buf, int len)
         rc = 0;
         goto out;
     }
+
+    rem_bytes = img_blen % flash_area_align(fap);
+
     if (curr_off + img_blen < img_size) {
-        rem_bytes = img_blen % flash_area_align(fap);
         if (rem_bytes) {
             img_blen -= rem_bytes;
+            rem_bytes = 0;
```

Comment:
The `rem_bytes` are needed only when `curr_off + img_blen >= img_size`, actually `==`, so this `if` could be reversed to just get `rem_bytes` only when the last part of image is written. Right?

Reply:
I think, that to be precisely all transfer should be aligned, but the last one must not.

Reply:
@de-nordic 
Introduced a check for the size-alignment of the rest of image blocks as
All of their sizes have to be aligned to
multiple of the flash write-block-size. This should be ensured
as well, so fare code has been cutting blocks to meet the requirement.